src/main/kotlin/club/chachy/kransformer/ClassTransformer.kt

```diff
@@ -42,25 +41,43 @@ open class ClassTransformer : IClassTransformer {
             node.accept(it)
         }
 
-        writerHandlers.forEach { it.invoke(writer) }
-
+        writerHandlers.forEach { it.invoke(WriterHandler(transformedName, writer)) }
         return writer.toByteArray()
     }
 
-    fun writerHandler(block: (ClassWriter) -> Unit) {
-        writerHandlers.add(block)
-    }
-
     /**
      * Load transformer block into a list mapped to the class name.
      *
      * @param name Name of the class to be transformed
      * @param transformer The transformer block where your transformations are.
      */
-
     fun transform(vararg name: String, transformer: TransformerBuilder.() -> Unit) {
         name.forEach {
             transformers.put(it, transformer)
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * Called after the class writing happens, this can be used like below to write .class files to disk
+     */
+    fun writerHandler(block: (WriterHandler) -> Unit) = writerHandlers.add(block)
+
+    init {
+        writerHandler {
```

Comment:
Maybe here could you create a variable called outputWriter which contains this implementation and add it directly to the list 

so val outputBytecode = { ... }
and in the list parameters add outputBytecode